### PR TITLE
Fixing issues and adding maven to manage dependencies

### LINKED ISSUE #3: Add a management and builder tool to the project
Sugestions: ant, maven or gradle.

### PR DESCRIPTION
**Pull request by carlospatinos**
Fix https://github.com/mdcdxcvi/Explorer/issues/3
I also fixed formatting of the code, and restructure the project to be maven compatible 
You can run mvn clean install to build the project, tests are run as default

### CODE REVIEW

**Comment by romavini** [comment]
The code has just been refactored. Can you fix your PR?

**Review by carlospatinos — COMMENTED** [review]
I moved all your new classes into the new structure.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.mdcdxcvi</groupId>
+  <artifactId>explorer</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>explorer</name>
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
```
Comment (romavini) [inline pom.xml]: Please remove the tag.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.mdcdxcvi</groupId>
+  <artifactId>explorer</artifactId>
+  <version>1.0-SNAPSHOT</version>
```
Comment (romavini) [inline pom.xml]: Leave just version 1.0, please.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.mdcdxcvi</groupId>
```
Comment (romavini) [inline pom.xml]: Replace tha package name to `io.github.mdcdxcvi`.

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  <version>1.0-SNAPSHOT</version>
+
+  <name>explorer</name>
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
+    <maven.compiler.target>1.7</maven.compiler.target>
```
Comment (romavini) [inline pom.xml]: Change to `1.8`.

File: src/main/java/com/mdcdxcvi/App.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.mdcdxcvi;
```
Comment (romavini) [inline src/main/java/com/mdcdxcvi/App.java]: Can you remove this file?

File: src/test/java/com/mdcdxcvi/AppTest.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.mdcdxcvi;
```
Comment (romavini) [inline src/test/java/com/mdcdxcvi/AppTest.java]: Remove this file too.

**Comment by romavini** [comment]
Update the `README.md` with the new way of running the project with Maven.

File: pom.xml (hunk 2 above)
Comment (carlospatinos) [reply pom.xml]: normally you use snapshot and then on every release you tag the specific version but ok, will do what you want

**Comment by carlospatinos** [comment]
All changes required are implemented. I think this covers the purpose of the issue, if you need further changes that are outside the scope probably better to open other issues

File: README.md [hunk 7]
```diff
@@ -16,6 +16,13 @@ Download the project from Github, compile using
...
 >>> javac MAP.java MONSTRO.java EXPLORER.java COMIDA.java GameScreenCli.java
 ```
+
+To install dependencies, maven and java 1.8 are needed, then you can run: 
+
+```sh
+mvn clean install 
+```
+
 -------
```
Comment (mdcdxcvi) [inline README.md:26]: Can you change to:
mvn clean compile package
java -jar target\explorer-1.0.jar

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.0</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.22.1</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-jar-plugin</artifactId>
+          <version>3.0.2</version>
```
Comment (romavini) [inline pom.xml:51]: Can you add this config?
```
<configuration>
    <archive>
        <manifest>
            <mainClass>io.github.mdcdxcvi.MAP</mainClass>
        </manifest>
    </archive>
</configuration>
```

File: README.md [hunk 9]
```diff
@@ -16,6 +16,13 @@ Download the project from Github, compile using
 ``` java 
 >>> javac MAP.java MONSTRO.java EXPLORER.java COMIDA.java GameScreenCli.java
 ```
+
+To install dependencies, maven and java 1.8 are needed, then you can run: 
+
+```sh
+mvn clean install 
+```
```
Comment (romavini) [inline README.md]: Can you change to:
```
mvn clean compile package
java -jar target\explorer-1.0.jar
```

File: src/main/java/io/github/mdcdxcvi/MAPA.java [hunk 10]
```diff
@@ -0,0 +1,523 @@
+package io.github.mdcdxcvi;
```
Comment (romavini) [inline src/main/java/io/github/mdcdxcvi/MAPA.java]: This file does not exist anymore in the repository. It has been modified to MAP.java

File: README.md [hunk 11]
```diff
@@ -16,6 +16,13 @@ Download the project from Github, compile using
 ``` java 
 >>> javac MAP.java MONSTRO.java EXPLORER.java COMIDA.java GameScreenCli.java
 ```
```
Comment (romavini) [inline README.md:18]: Remove the old instructions.

**Review by romavini — COMMENTED** [review]
I'm really glad for your contribution, but still, some things to be fixed.

**Comment by carlospatinos** [comment]
I think the original scope has changed since, I dont mind supporting but i think we can keep going on and on, maybe better to stick with what is really needed and then if more improvements are required create other issues?

**Comment by romavini** [comment]
Of course, I understand. Just ask for these final fixes because the game is not working as it should.
```
Out[]: no main manifest attribute, in explorer-1.0.jar
```

**Comment by carlospatinos** [comment]
Last changes sent

**Comment by romavini** [comment]
Thanks for your contribution! ^^

**Pull request merged**